### ISSUE TITLE
Installation process

### ISSUE THREAD

**Issue by lmz** [BayAreaMetro/tm2py#72]
I'm not sure if this is an chore or a bug yet so I'm just starting a discussion here.  My tm2py environment broke (GDAL runtime errors) so I decided to make a new one.

I started by created a new conda environment
and installing python and then installing Emme packages (from the Emme GUI)
and then installing tm2py:

```
(base) X:\tm2py>conda create --name tm2py
(base) X:\tm2py>conda activate tm2py
(tm2py) X:\tm2py>conda install python
```
At this point, I installed Emme python packages via the Emme GUI (Application Options > Modeller)
```
(tm2py) X:\tm2py>pip show inro-emme
Name: inro-emme
Version: 4.6.0
Summary: Emme API
Home-page: http://inrosoftware.com
Author:
Author-email:
License: Proprietary
Location: c:\program files\inro\emme\emme 4\emme-4.6.0\python37\lib\site-packages
Requires: Cheetah3, colour, numexpr, numpy, orderedset, ply, pywin32
Required-by: inro-emme-agent, inro-modeller

(tm2py) X:\tm2py>pip install -e .
Obtaining file:///X:/tm2py
  Preparing metadata (setup.py) ... done
Collecting geojson
  Using cached geojson-2.5.0-py2.py3-none-any.whl (14 kB)
Collecting geographiclib
  Using cached geographiclib-2.0-py3-none-any.whl (40 kB)
Collecting geopandas>0.8.0
[... lots of info not pasted for brevity...]

  Attempting uninstall: pytz
    Found existing installation: pytz 2018.9
    Uninstalling pytz-2018.9:
ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'c:\\program files\\inro\\emme\\emme 4\\emme-4.6.0\\python37\\lib\\site-packages\\pytz-2018.9.dist-info\\DESCRIPTION.rst'
Consider using the `--user` option or check the permissions.
```

**Comment by lmz** [comment]
cc @inrokevin @e-lo

**Comment by lmz** [comment]
I was trying this order of operations because if I install using the directions (the environment file) then that's how I got the GDAL error since it installed GDAL instead of using the Emme version:
```
(tm2py) X:\tm2py>pip show gdal
Name: GDAL
Version: 2.3.3
Summary: GDAL: Geospatial Data Abstraction Library
Home-page: http://www.gdal.org
Author: Frank Warmerdam
Author-email: warmerdam@pobox.com
License: MIT
Location: c:\program files\inro\emme\emme 4\emme-4.6.0\python37\lib\site-packages
Requires:
Required-by: cityphi-emme, cityphi-studio, inro-emme-agent
```

**Comment by lmz** [comment]
After a lot of iterations, the following commands resolved errors which *seem* to be related to packages:
```
(base) X:\tm2py>conda create --name tm2py
(base) X:\tm2py>conda activate tm2py
(tm2py) X:\tm2py>conda install python
(tm2py) conda install pytz==2021.1
(tm2py) conda install shapely==1.8.0
(tm2py) conda install python-dateutil==2.8.2
(tm2py) conda install pyproj==3.3.1
(tm2py) conda install pygments==2.12.0
(tm2py) conda install markupsafe==2.1.0
(tm2py) conda install jinja2==3.0.3
(tm2py) conda install click==8.1.0
(tm2py) conda install attrs==21.4.0
(tm2py) conda install pandas==1.4.2
(tm2py) conda install matplotlib-base==3.5.1 matplotlib-inline==0.1.3
(tm2py) conda install cligj==0.7.2
```
Installed Emme python packages via the Emme GUI (Application Options > Modeller)
```
(tm2py) X:\tm2py>pip install -e .
```

But now I'm getting this error when I try to run on develop branch:
```
(tm2py) E:\Model2C-Share\Projects_TM2\example_union_test_highway>python X:\tm2py\bin\tm2py -m model_config.toml -s scenario_config.toml
Traceback (most recent call last):
  File "X:\tm2py\bin\tm2py", line 5, in <module>
    from tm2py.controller import RunController
  File "\\mainmodel\mainmodelshare\tm2py\tm2py\__init__.py", line 12, in <module>
    from .controller import RunController
  File "\\mainmodel\mainmodelshare\tm2py\tm2py\controller.py", line 22, in <module>
    from tm2py.emme.manager import EmmeManager
  File "\\mainmodel\mainmodelshare\tm2py\tm2py\emme\manager.py", line 19, in <module>
    from inro.emme.database.emmebank import Emmebank
ImportError: bad magic number in 'inro.emme.database.emmebank': b'B\r\r\n'
```

**Comment by lmz** [comment]
I updated the above comment because I got further, but I still get an exception from Emme.  @inrokevin ?

**Comment by e-lo** [comment]
Just a note that in addition to updating the documentation, we should create a conda `environment.yml` file which is like a lock file of "what we know works"

**Comment by xyzjayne** [comment]
This might be a separate issue, but the ``environment.yml`` file doesn't appear to be working for a fresh install. It appears that most of the packages could not be identified in conda (an incomplete list below).

![image](https://user-images.githubusercontent.com/25780417/171930682-7e8bdcaa-2ff3-482b-89d0-78bf3f3b58e5.png)

**Comment by i-am-sijia** [comment]
From internal meeting on June 12, 2023 (cc @DavidOry) - Jia Hao was able to finish installing tm2py and start a run. So currently both Arash and Jia Hao are unblocked by the installation.

@AshishKuls @arashasadabadi are creating a memo documenting the latest installation steps. @AshishKuls, please update the installation in ReadMe.md.
